packages/compass-e2e-tests/tests/auto-update.test.ts

```diff
@@ -13,66 +13,94 @@ function wait(ms: number) {
+        // normal circumstances because it is destructive - it overwrites Compass
+        // itself
+        this.skip();
       }
-    } finally {
-      await browser.screenshot(screenshotPathName('auto-update-from'));
-      await cleanup(compass);
-    }
 
-    if (process.env.AUTO_UPDATE_UPDATABLE === 'true') {
-      console.log(
-        'pause to make sure the app properly exited before starting again'
-      );
-      await wait(10_000);
-
-      console.log('starting compass a second time');
-      // run the app again and check that the version changed
-      const compass = await init('auto-update from restart', {
-        firstRun: false,
-      });
+      // run the app and wait for it to auto-update
+      console.log('starting compass the first time');
+      const compass = await init(testName, { firstRun: true });
       const { browser } = compass;
       try {
         await browser.$(Selectors.AutoUpdateToast).waitForDisplayed();
-        await browser
-          .$(Selectors.AutoUpdateReleaseNotesLink)
-          .waitForDisplayed();
+
+        if (process.env.AUTO_UPDATE_UPDATABLE === 'true') {
+          const restartButton = browser.$(Selectors.AutoUpdateRestartButton);
+          await restartButton.waitForDisplayed();
+
+          if (process.env.EXPECTED_UPDATE_VERSION) {
+            expect(
+              await browser.$(Selectors.AutoUpdateToast).getText()
+            ).to.contain(
+              `Compass is ready to update to ${process.env.EXPECTED_UPDATE_VERSION}!`
+            );
+          }
+
+          // We could click the restart button to apply the update and restart the
+          // app, but restarting the app confuses webdriverio or at least our test
+          // helpers. So we're going to just restart the app manually.
+          await browser.pause(1000);
+        } else {
+          // When auto-update is not supported the toast contains a link to
+          // download
+          const linkElement = browser.$(Selectors.AutoUpdateDownloadLink);
+          await linkElement.waitForDisplayed();
+          expect(await linkElement.getAttribute('href')).to.equal(
+            'https://www.mongodb.com/try/download/compass?utm_source=compass&utm_medium=product'
+          );
+
+          if (process.env.EXPECTED_UPDATE_VERSION) {
+            expect(
+              await browser.$(Selectors.AutoUpdateToast).getText()
+            ).to.contain(
+              `Compass ${process.env.EXPECTED_UPDATE_VERSION} is available`
+            );
+          }
+        }
       } finally {
-        await browser.screenshot(
-          screenshotPathName('auto-update-from-restart')
-        );
+        await browser.screenshot(screenshotPathName(testName));
         await cleanup(compass);
       }
-    }
-  });
+
+      if (process.env.AUTO_UPDATE_UPDATABLE === 'true') {
+        console.log(
+          'pause to make sure the app properly exited before starting again'
+        );
+        await wait(10_000);
```

Comment:
Could we poll the `ps` with some `pid` instead?  🤔 Ideally from within the `cleanup` function.

Reply:
Yeah, but where do we get the pid? I think we can clean it up but in an effort to get somewhere let's do that as a follow-up.